server/src/main/java/org/elasticsearch/common/unit/ByteSizeValue.java

```diff
@@ -292,7 +292,7 @@ private static ByteSizeValue parse(
             } catch (final NumberFormatException e) {
                 try {
                     final double doubleValue = Double.parseDouble(s);
-                    DeprecationLoggerHolder.deprecationLogger.warn(
+                    DeprecationLoggerHolder.deprecationLogger.critical(
```

Comment:
<sigh> things have evolved since I wrote this. We're not outlawing fractional bytes entirely anymore.